File: benchmarks/vm/15_break_continue_loop.rb
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
total = 0
seen = 0
(0...300_000).each do |i|
  break if i > 123_456
  next if (i % 17) == 0

  seen += 1
  if (i % 5) == 0
    total += i * 2
  else
    total += i
  end
end
score = total + (seen * 11)
total_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000
puts "seen = #{seen}"
puts "score = #{score}"
puts format("BENCH_RESULT: %.3f ms", total_ms)

File: benchmarks/vm/15_break_continue_loop.lua
local t0 = os.clock()
local total = 0
local seen = 0
for i = 0, 299999 do
    if i > 123456 then
        break
    end
    if (i % 17) ~= 0 then
        seen = seen + 1
        if (i % 5) == 0 then
            total = total + (i * 2)
        else
            total = total + i
        end
    end
end
local score = total + (seen * 11)
local total_ms = (os.clock() - t0) * 1000
print("seen = " .. tostring(seen))
print("score = " .. tostring(score))
print(string.format("BENCH_RESULT: %.3f ms", total_ms))
